src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2407,6 +2407,13 @@ TargetShardIntervalForFastPathQuery(Query *query, Const **partitionValueConst,
 		PruneShards(relationId, relationIndex, make_ands_implicit((Expr *) quals),
 					&queryPartitionValueConst);
 
+	if (PartitionMethod(relationId) != DISTRIBUTE_BY_NONE &&
```

Comment:
Can we have an early exit for reference tables? It seems odd that we made it all the way here and it's not immediately obvious how this relates to queryPartitionValueConst.